File: 75-sort-colors/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        n = len(nums)
        low = 0
        high = n - 1
        mid = 0
        while mid <= high:
            if nums[mid] < 1:
                nums[low], nums[mid] = nums[mid], nums[low]
                low += 1
                mid += 1
            elif nums[mid] == 1:
                mid += 1
            else:
                nums[mid], nums[high] = nums[high], nums[mid]
                high -= 1

File: 75-sort-colors/sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int low, mid = 0;
        int high = nums.size() - 1;
        while (mid <= high) {
            if (nums[mid] == 0) {
                nums[mid] = nums[low];
                nums[low] = 0;
                low += 1;
                mid += 1;
            }
            else if (nums[mid] == 1) {
                mid += 1;
            }
            else {;
                nums[mid] = nums[high];
                nums[high] = 2;
                high -= 1;
            }
        }
    }
};